### PR TITLE
New Configure Workflow Menu UI

### PR DESCRIPTION
**Pull request by anuprulez**
# Introduction:

This Pull Request replaces "configure_menu.mako" by JavaScript. It is another step in the line of the https://github.com/galaxyproject/galaxy/pull/4047 to remove mako files for Workflows.

# User Interface:
configure_menu

# Details:
It merges both the types of workflows (user's own and shared) into one table which differ by the text in the "owner" column. Again, the workflows are shown in the center panel. After saving the preferences, user is redirected to workflow home screen. As an enhancement, the workflow's table has two buttons (top right) for checking and unchecking all workflows before saving.

# Credits:
Thanks a lot @bgruening 

All suggestions are welcome. Thank you!

### CODE REVIEW

**Comment by bgruening** [comment]
Thanks a lot @anuprulez!

**Comment by martenson** [comment]
@galaxybot test this

File: lib/galaxy/webapps/galaxy/api/workflows.py [hunk 1]
```diff
@@ -53,7 +53,53 @@ def __get_full_shed_url( self, url ):
...
+        return self.get_workflows_list( trans, False, kwd )
 
+    @expose_api
+    def configure_workflow_menu( self, trans, **kwd ):
+        """
+        GET /api/workflows/configure_workflow_menu
+        """
+        payload = kwd.get( 'payload' )
+        user = trans.get_user()
+        if trans.request.method == "PUT":
```
Comment (guerler) [inline lib/galaxy/webapps/galaxy/api/workflows.py]: Methods should have their own function. I would also suggest to name the endpoint just `workflow_menu`.

**Comment by martenson** [comment]
@galaxybot test this

**Comment by anuprulez** [comment]
@guerler Is it ok now to be merged? thanks :)

**Comment by guerler** [comment]
Minor but I think the endpoint should be 'workflow/menu' where 'PUT' saves settings and a 'GET' returns the settings. So the url should always be `/api/workflows/menu`. The functions can be named set_/get_ just like for the user endpoints. Sorry if I was not clear before.

**Comment by guerler** [comment]
@anuprulez please checkout the comment above. Everything else looks very nice to me. Pretty cool. Thanks a lot for your contributions.

**Comment by anuprulez** [comment]
@guerler Yes, I have made the required changes. Thanks a lot for the review :)

**Comment by anuprulez** [comment]
@guerler does it look ok now? thanks :)

**Comment by anuprulez** [comment]
@guerler Thank you for the changes and sorry for not understanding the changes in the way you wished. 

One doubt I have may be for future reference. How is it beneficial to have one api url (/api/workflows/menu) for two api methods/calls (get and set here) instead of having one api url and one method/call (like the way I did before)? Is it required for api tests during build or do users directly access the api through the url? 

Thanks!

**Comment by guerler** [comment]
I think `/api/workflows/menu` is fine. Its somewhat restful. GET requests data and PUT saves it to the entity which is a menu.

**Comment by bgruening** [comment]
@guerler can we merge this?

**Comment by dannon** [comment]
Just catching up here, and it's not a major change really (nor required, just throwing the idea out there), but to me this might make more sense under /api/user/preferences or the like.  (That is, it seems it's more a user preference than an attribute of workflows at its core -- it's a purely application display artifact)

**Comment by guerler** [comment]
I agree.

**Comment by bgruening** [comment]
@dannon  @guerler should this be changed before the merge? Or can we change this later. @anuprulez has already more to come.

**Comment by guerler** [comment]
I think it can be changed in a follow-up if @dannon is ok with that.

**Comment by dannon** [comment]
Generally speaking I think we should probably nail down the endpoints first, but I'm fine with it being a followup in this case -- we just need to have the API endpoint in its final form before the next release.

May have just run into a problem with security on this one though, hang on a sec.

**Comment by dannon** [comment]
(following up with separate PR as the security bit is unrelated to this specific code -- just this feature)

**Comment by anuprulez** [comment]
@dannon Thanks for merging. Will follow up the changes suggested in the next PR

**Comment by anuprulez** [comment]
@dannon Do you wish to change just the url to `/api/user/preferences`? This will involve moving the api calls (which make call to other methods defined in WorkflowsAPIController controller to fetch a list of workflows) to `/users` api file. So, changing the url would involve not only the change in urls but moving some methods to `/users` api file. Is it ok? 
@bgruening

**Pull request merged**